src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -585,8 +576,28 @@ class IntervalExpressionSegment(BaseSegment):
     ParameterNameSegment=RegexParser(
         r"`?[A-Za-z0-9_]*`?", CodeSegment, name="parameter", type="parameter"
     ),
-    SingleIdentifierGrammar=ansi_dialect.get_grammar("SingleIdentifierGrammar").copy(
-        insert=[Ref("SessionVariableNameSegment")]
+    SingleIdentifierGrammar=OneOf(
+        Ref("SessionVariableNameSegment"),
+        # https://dev.mysql.com/doc/refman/8.0/en/account-names.html
+        # https://dev.mysql.com/doc/refman/8.0/en/role-names.html
+        Sequence(
+            OneOf(
+                Ref("NakedIdentifierSegment"),
+                Ref("QuotedIdentifierSegment"),
+                Ref("SingleQuotedIdentifierSegment"),
+                Ref("DoubleQuotedLiteralSegment"),
+            ),
+            Sequence(
+                Ref("AtSignLiteralSegment"),
+                OneOf(
+                    Ref("NakedIdentifierSegment"),
+                    Ref("QuotedIdentifierSegment"),
+                    Ref("SingleQuotedIdentifierSegment"),
+                    Ref("DoubleQuotedLiteralSegment"),
+                ),
+                optional=True,
+            ),
```

Comment:
If this sequence is only valid for account names and roles is it really necessary to put them everywhere a Single Identifier is valid?

Would, it not be better to create a new Grammar (or Grammars) for these and just reference them in any `ObjectReferenceSegment` instead? Or, better yet, update `AccessStatementSegment` to reference them?

Reply:
Agree that it would be better to limit the scope. In addition to granting access, this same thing exists for `CREATE USER` (and maybe more): https://dev.mysql.com/doc/refman/8.0/en/create-user.html . That makes me think `ObjectReferenceSegment` might be best?

I wasn't sure how to override `AccessStatementSegment`. Ideally I'd want to keep all the complexity already in ANSI and just change this one part:
https://github.com/sqlfluff/sqlfluff/blob/3de129150dc758fac709fde51f926bb34ce7e9ea/src/sqlfluff/dialects/dialect_ansi.py#L3032
Does that one part become a grammar in ANSI that gets overwritten in MySQL? Or is there a different way to change this one part for MySQL?

Reply:
Yeah pulling a piece of to its own Grammar and overriding just that is the best way to have a small change to a more complex statement.

The alternative would be to build this in `ObjectReferenceSegment` instead of `SingleIdentifierGrammar`? That would at least limit the scope a little (and prevent the duplication currently in this PR).

Or we could just merge as is. It works and isn’t completely without precedent as a few of the dialects have more complex structures in `SingleIdentifierGrammar`.

WDYT? I think I’m veering towards the middle section (defining this in `ObjectReferenceSegment`

Reply:
I'll give `ObjectReferenceSegment` a shot 👍

Reply:
I think the resulting changes to `x_definer.yml` parsings is actually better after https://github.com/sqlfluff/sqlfluff/pull/3286/commits/da6143e9fb8628c86b253d993234332bcd9d7350